LazyInitializationException thrown by scheduled notification jobs trying to get roles for user profiles

ERROR [23-January 22:24:05] [schedulerFactoryBean_Worker-5] org.quartz.core.JobRunShell  - Job EMAIL.RESOURCE_REMOVED_FROM_TASK threw an unhandled Exception:
org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: org.libreplan.business.users.entities.User.profiles, could not initialize proxy - no Session
        at org.hibernate.collection.internal.AbstractPersistentCollection.throwLazyInitializationException(AbstractPersistentCollection.java:584)
        at org.hibernate.collection.internal.AbstractPersistentCollection.withTemporarySessionIfNeeded(AbstractPersistentCollection.java:201)
        at org.hibernate.collection.internal.AbstractPersistentCollection.initialize(AbstractPersistentCollection.java:563)
        at org.hibernate.collection.internal.AbstractPersistentCollection.read(AbstractPersistentCollection.java:132)
        at org.hibernate.collection.internal.PersistentSet.iterator(PersistentSet.java:163)
        at org.libreplan.business.users.entities.User.isInRole(User.java:208)
        at org.libreplan.importers.notifications.ComposeMessage.composeMessageForUser(ComposeMessage.java:139)
        at org.libreplan.importers.notifications.realization.SendEmailOnResourceRemovedFromTask.composeMessageForUser(SendEmailOnResourceRemovedFromTask.java:79)
        at org.libreplan.importers.notifications.realization.SendEmailOnResourceRemovedFromTask.sendEmail(SendEmailOnResourceRemovedFromTask.java:69)
        at org.libreplan.importers.notifications.jobs.SendEmailOnResourceRemovedFromTaskJob.executeInternal(SendEmailOnResourceRemovedFromTaskJob.java:47)
        at org.springframework.scheduling.quartz.QuartzJobBean.execute(QuartzJobBean.java:75)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
ERROR [23-January 22:24:05] [schedulerFactoryBean_Worker-5] org.quartz.core.ErrorLogger  - Job (EMAIL.RESOURCE_REMOVED_FROM_TASK threw an exception.
org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: org.libreplan.business.users.entities.User.profiles, could not initialize proxy - no Session]
        at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: org.libreplan.business.users.entities.User.profiles, could not initialize proxy - no Session
        at org.hibernate.collection.internal.AbstractPersistentCollection.throwLazyInitializationException(AbstractPersistentCollection.java:584)
        at org.hibernate.collection.internal.AbstractPersistentCollection.withTemporarySessionIfNeeded(AbstractPersistentCollection.java:201)
        at org.hibernate.collection.internal.AbstractPersistentCollection.initialize(AbstractPersistentCollection.java:563)
        at org.hibernate.collection.internal.AbstractPersistentCollection.read(AbstractPersistentCollection.java:132)
        at org.hibernate.collection.internal.PersistentSet.iterator(PersistentSet.java:163)
        at org.libreplan.business.users.entities.User.isInRole(User.java:208)
        at org.libreplan.importers.notifications.ComposeMessage.composeMessageForUser(ComposeMessage.java:139)
        at org.libreplan.importers.notifications.realization.SendEmailOnResourceRemovedFromTask.composeMessageForUser(SendEmailOnResourceRemovedFromTask.java:79)
        at org.libreplan.importers.notifications.realization.SendEmailOnResourceRemovedFromTask.sendEmail(SendEmailOnResourceRemovedFromTask.java:69)
        at org.libreplan.importers.notifications.jobs.SendEmailOnResourceRemovedFromTaskJob.executeInternal(SendEmailOnResourceRemovedFromTaskJob.java:47)
        at org.springframework.scheduling.quartz.QuartzJobBean.execute(QuartzJobBean.java:75)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        ... 1 more

--- a/libreplan-business/src/main/resources/org/libreplan/business/users/entities/Users.hbm.xml
+++ b/libreplan-business/src/main/resources/org/libreplan/business/users/entities/Users.hbm.xml
@@ -53,7 +53,7 @@
         </set>
 
         <!-- Index created in a database-object section -->
-        <set name="profiles" table="user_profiles">
+        <set name="profiles" table="user_profiles" lazy="false">
             <key column="user_id"/>
             <many-to-many column="profile_id" class="org.libreplan.business.users.entities.Profile"/>
         </set>

